### PR TITLE
[9/x][dynamic child loading] Remove old child object logic

### PR DESCRIPTION
**Pull request by tnowacki**
- Removed transfer_to_object
- Removed child objects as transaction arguments
- Updated examples and tests

- Blocked by #5454

### CODE REVIEW

File: crates/sui-core/src/transaction_input_checker.rs [hunk 1]
```diff
@@ -153,11 +153,11 @@ async fn check_objects(
     // TODO: We should be able to allow the same shared object to show up
     // in more than one SingleTransactionKind. We need to ensure that their
     // version number only increases once at the end of the Batch execution.
-    let mut owned_object_authenticators: HashSet<SuiAddress> = HashSet::new();
+    let mut used_objects: HashSet<SuiAddress> = HashSet::new();
```
Comment (lxfind) [inline crates/sui-core/src/transaction_input_checker.rs:156]: First part of the comment could be removed.

**Review by lxfind — APPROVED** [review]

**Comment by lzpap** [comment]
@tnowacki Now that `transfer_to_object` is removed, is it still possible to transfer an object to another object? If yes, how?

Suppose I have `Object A` and `Object B`. I know the `ID` of `Object B`, but I don't own it/can't touch it.
Can I still send `Object A` to be owned by `Object B`? 

Thanks for the answer!

**Comment by tnowacki** [comment]
> @tnowacki Now that `transfer_to_object` is removed, is it still possible to transfer an object to another object? If yes, how?
> 
> Suppose I have `Object A` and `Object B`. I know the `ID` of `Object B`, but I don't own it/can't touch it. Can I still send `Object A` to be owned by `Object B`?
> 
> Thanks for the answer!

You usecase would still work with `dynamic_object_fields`. Check out the migration guide: https://sui.io/resources-sui/dynamic-fields-migration-guide/

We recommend giving some deeper thought to how you want to relate the objects, but the migration guide does give a drop in replacement for quick migration :)

**Comment by lzpap** [comment]
Thank you for the quick answer @tnowacki!

I've seen the migration guide, but I still don't fully understand how my use case would work. If I want to add a `dynamic_object_field` to an object, the library API is the following:
https://github.com/MystenLabs/sui/blob/058f1b86ae734cdac852435a4ab42ccbad1996ac/crates/sui-framework/sources/dynamic_object_field.move#L38-L51

I have to pass in a mutable reference to the _target_ object (`object: &mut UID` ), therefore I either have to own _target_ or it must be a shared one. In any case, _target_ object will be consumed and updated in the transaction that calls this API.
 - Is it possible to send something to an object without having to explicitly consume/update it?
 - Can `Object ID`s function similarly to addresses in a sense that an object owned by an `Object ID` can only be referenced in a tx that passes in the actual object for `Object ID`?
 - I understand that `dynamic_object_fields` serve a similar purpose, but as discussed above, you have to own the target object (as far as I can tell) to be able to send something to it, while in case of addresses, it is not the case. I can send anything to your address even though I don't own it.

**Pull request merged**